wasm_src/ast_wrapper/post.ts

```diff
@@ -197,6 +198,37 @@ Module.transformPointArrays = function (wcsInfo: number, xIn: Float64Array, yIn:
     return result;
 };
 
+Module.transform3DPointArrays = function (wcsInfo: number, xIn: Float64Array, yIn: Float64Array, zIn: Float64Array, forward: boolean = true) {
+    // Return empty array if arguments are invalid
+    if (!(xIn instanceof Float64Array) || !(yIn instanceof Float64Array) || !(zIn instanceof Float64Array) || xIn.length !== yIn.length || yIn.length !== zIn.length) {
+        return {x: new Float64Array(1), y: new Float64Array(1), z: new Float64Array(1)};
+    }
+
+    // Allocate and assign WASM memory
+    const N = xIn.length;
+    const xInPtr = Module._malloc(N * 8);
```

Comment:
from above, something like

```javascript
const inPtr = Module._malloc(3 * N * 8);
const xInPtr = inPtr;
const yInPtr = xInPtr + 8 * N;
const zInPtr = zInPtr + 8 * N;
```
would assign the right amount of memory and then copy the x, y, and z arrays into it. You could then simply pass `inPtr` in to `transform3DArray` directly

Reply:
fixed.